py3status/modules/playerctl.py

```diff
@@ -0,0 +1,268 @@
+        self.py3.update()
+
+    def _on_metadata(self, player, metadata, manager):
+        self.py3.update()
+
+    def _status_changed(self, player, status, manager):
+        self.py3.update()
+
+    def _get_player_position(self, player):
+        try:
+            # Playerctl doesn't support getting positions for all players
+            microseconds = player.get_position()
+            position = self._microseconds_to_time(microseconds)
+        except GLib.GError:
+            position = None
+
+        return position
+
+    @staticmethod
+    def _microseconds_to_time(microseconds):
+        seconds = microseconds // 1_000_000
+        m, s = divmod(seconds, 60)
+        h, m = divmod(m, 60)
+        time = f"{h}:{m:02d}:{s:02d}"
+        return time.lstrip("0").lstrip(":")
+
+    def _set_color(self, player, data):
+        """Set the color and the control variables for a player"""
+        status = player.props.playback_status
+        if status == Playerctl.PlaybackStatus.PLAYING:
+            data["color"] = self.color_playing
+            data["is_playing"] = True
+        elif status == Playerctl.PlaybackStatus.PAUSED:
+            data["color"] = self.color_paused
+            data["is_paused"] = True
+        else:
+            data["color"] = self.color_stopped
+            data["is_stopped"] = True
+
+    def _set_data_from_metadata(self, player, data):
+        """Set any data retrieved directly from the metadata for a player"""
+        metadata = dict(player.props.metadata)
+
+        duration_ms = metadata.get("mpris:length")
+        if duration_ms:
+            data["duration"] = self._microseconds_to_time(duration_ms)
+        else:
+            data["duration"] = None
+
+    def _get_player_data(self, player):
+        data = {}
+
+        data["album"] = player.get_album()
+        data["artist"] = player.get_artist()
+        data["title"] = player.get_title()
+        data["player"] = player.props.player_name
+
+        self._set_color(player, data)
+
+        data["position"] = self._get_player_position(player)
+        self._set_data_from_metadata(player, data)
+
+        return data
+
+    def playerctl(self):
+        tracked_players = self.manager.props.players
+        cached_until = self.cache_timeout if tracked_players else self.sleep_timeout
+
+        players = []
+        for player in tracked_players:
+            player_data = self._get_player_data(player)
+
+            # Delete after referencing because it shouldn't be a valid placeholder
+            color = player_data["color"]
+            del player_data["color"]
+
+            format_player = self.py3.safe_format(self.format_player, player_data)
+            self.py3.composite_update(format_player, {"index": player_data["player"]})
+            self.py3.composite_update(format_player, {"color": color})
+            players.append(format_player)
+
+        format_separator = self.py3.safe_format(self.format_separator)
```

Comment:
(Silly) Please rename to `format_player_separator`.

If there is more than 1 `format_` loops (such as `format_player`, `format_song`, `format_tag`, whatever) in this module (or any other module), having a single separator config to control two or more separators can be unwanted... so separators for all `format_` loops ought to be kept separately too.